TombEngine/Objects/TR3/Entity/tr3_trex.cpp

```diff
@@ -168,18 +169,20 @@ namespace TEN::Entities::Creatures::TR3
 			case TREX_STATE_ATTACK:
 				if (item->TouchBits.Test(TRexAttackJoints))
 				{
-					item->Animation.TargetState = TREX_STATE_KILL;
-
-					DoDamage(LaraItem, 1500);
-					LaraTRexDeath(item, LaraItem);
+					CreatureKill(item, TREX_ANIM_KILL, LEA_TREX_DEATH, TREX_STATE_KILL, LS_DEATH);
+					Camera.targetDistance = BLOCK(3);
 				}
 
-				item->Animation.RequiredState = TREX_STATE_WALK_FORWARD;
+				break;
+
+			case TREX_STATE_KILL:
+				headYRot = 0;
```

Comment:
Why `headYRot` and `MaxTurn` are zeroed?

Reply:
MaxTurn for same reason, to prevent the Trex from rotating while eating Lara.

The head has is also for the same. The Trex rotates its head to look at Lara, but if it started the killing animation with the rotated head, because the new state, it was not refreshing so in some cases, the raptor did the killing animation with the head looking to a side (what didn't fit with Lara's position). I made that late change to avoid that visual bug.